### PR TITLE
[Nfs] Add nfs scale with spec storage

### PR DESCRIPTION
**Pull request by pranavprakash20**
# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Review by MohitBis — DISMISSED** [review]
LGTM

File: tests/nfs/nfs_verify_scale.py [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+        for th in threads:
+            th.join()
+        log.info("Completed running IO on all mounts")
+
+    except Exception as e:
+        log.error(f"Failed to perform scale ops : {e}")
+        return 1
+    finally:
+        log.info("Cleaning up")
+        cleanup_cluster(clients, nfs_mount, nfs_name, nfs_export)
```
Comment (MohitBis) [inline tests/nfs/nfs_verify_scale.py:164]: Please update cleanup operation to support multiple export

**Comment by pranavprakash20** [comment]
run log:  https://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-44IGWK/

File: tests/nfs/nfs_verify_scale.py (hunk 1 above)
Comment (pranavprakash20) [reply tests/nfs/nfs_verify_scale.py:174]: Noted, all the mounts and exports are now cleared

**Comment by pranavprakash20** [comment]
https://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-OID6I7/

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by msainiRedhat — DISMISSED** [review]
LGTM

**Comment by pranavprakash20** [comment]
Run log for 200 exports:  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-BZ5WEM/

File: tests/nfs/nfs_verify_scale.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
+import json
+from threading import Thread
+
+from nfs_operations import cleanup_cluster
+
+from cli.ceph.ceph import Ceph
+from cli.exceptions import ConfigError, OperationFailedError
+from cli.io.spec_storage import SpecStorage
```
Comment (vamahaja) [inline tests/nfs/nfs_verify_scale.py:8]: With new changes, this should be now from `cli.io import IO`

File: tests/nfs/nfs_verify_scale.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    primary_client,
+    benchmark,
+    load,
+    incr_load,
+    num_runs,
+    clients,
+    nfs_mount,
+    benchmark_defination,
+):
+    if SpecStorage(primary_client).run_spec_storage(
```
Comment (vamahaja) [inline tests/nfs/nfs_verify_scale.py:25]: Update according to new changes

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/nfs/nfs_verify_scale.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                # Perform IO
+                if execution_type == "serial":
+                    run_spec_storage_io(
+                        client,
+                        benchmark,
+                        load,
+                        incr_load,
+                        num_runs,
+                        clients,
+                        mount,
```
Comment (MohitBis) [inline tests/nfs/nfs_verify_scale.py:129]: Change to nfs_mount "/mnt/nfs"

**Review by msainiRedhat — APPROVED** [review]
LGTM

**Review by MohitBis — APPROVED** [review]
LGTM

**Review by vamahaja — APPROVED** [review]

**Pull request merged**